src/gwt/tools/generate-prefs.R

```diff
@@ -177,30 +183,91 @@ generate <- function (schemaPath, className) {
          "   /**\n",
          "    * ", def[["description"]], "\n",
          "    */\n")
-      
-      # Add a Java accessor for the preference, and an entry for syncing it with another copy
+
+      # Create a Java/GWT Properties file header for the preference
+      commentProperties <- paste0(
+         "# ", def[["description"]], "\n")
+
+      # Add title and description entries for the java constants file
       prefTitle <- if (is.null(def[["title"]])) "" else def[["title"]]
+      javaConstants <- paste0(javaConstants,
+         comment,
+         "   @DefaultStringValue(\"", prefixDefault, prefTitle, "\")\n",
+         "   String ", camel, "Title", "();\n",
+         "   @DefaultStringValue(\"", prefixDefault, def[["description"]], "\")\n",
+         "   String ", camel, "Description", "();\n")
+
+      if (!is.null(def[["enumReadable"]]))
+      {
+         javaConstants <- paste0(javaConstants,
+                                 paste0(mapply(function (enumVal, enumReadable) {
+                                    paste0(
+                                       "   @DefaultStringValue(\"", prefixDefault, enumReadable, "\")\n",
+                                       "   String ", camel, "Enum_", gsub("[^A-Za-z0-9_]", "_", enumVal), "();"
+                                    )
+                                 }, def[["enum"]], def[["enumReadable"]]), collapse = "\n"
+                                 ), "\n"
+         )
+      }
+
+      javaConstants <- paste0(javaConstants, "\n")
+
+      # Add title and description entries for the java properties file
+      javaProperties <- paste0(javaProperties,
+         commentProperties,
+         camel, "Title = ", prefixProperties, prefTitle, "\n",
+         camel, "Description = ", prefixProperties, def[["description"]], "\n")
+
+      if (!is.null(def[["enumReadable"]]))
```

Comment:
I think this would be a good place to validate that enumReadble has the same length as the enum.

Reply:
I've added a check near the beginning of the loop, to prevent unnecessary work from being done. This is under the assumption that if a pref has an invalid attribute, a msg should be logged, the generation of that pref should be skipped, and the script should continue with the next pref. Let me know if some other behaviour is desired! (i.e. halting the whole script)